File: src/main/java/com/team/puddy/domain/user/dto/request/RegisterUserRequest.java
package com.team.puddy.domain.user.dto.request;

import lombok.Builder;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;

public record RegisterUserRequest(@NotBlank String account,
                                  @NotBlank String password,
                                  @NotBlank String username,
                                  @NotBlank(message = "이메일을 입력해주세요")
                                  @Email(message = "이메일 형식이 아닙니다.") String email,
                                  boolean isNotificated ) {

    @Builder
    public RegisterUserRequest {}
}

File: src/main/kotlin/com/team/puddy/domain/user/dto/request/RegisterUserRequest.kt
package com.team.puddy.domain.user.dto.request;

import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank

data class RegisterUserRequest(
    @NotBlank val account: String,
    @NotBlank val password: String,
    @NotBlank(message = "이메일을 입력해주세요")
    @Email(message = "이메일 형식이 아닙니다.")
    val email: String,
    val isNotificated: Boolean,
)
